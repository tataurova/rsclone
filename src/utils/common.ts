export const extend = (a, b) => ({ ...a, ...b });

export const helperForTranslate = (value) => {
    switch (value) {
        case 'Резерв':
            return 'Reserve';
        case 'Поиск на месте':
            return 'Local search';
        case 'Стоп. Проверка информации':
            return 'Stop. Information verification';
        case 'Архив':
            return 'Archive';
    }
}

export const mockFunction = () => {
    // do nothing
};
