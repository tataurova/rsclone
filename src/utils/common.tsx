export const extend = (a, b) => ({ ...a, ...b });
