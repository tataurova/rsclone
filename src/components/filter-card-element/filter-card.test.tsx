import * as React from 'react';
import * as renderer from 'react-test-renderer';
import FilterCard from './filter-card';
import { mockFunction } from '../../utils/common';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

describe('<FilterCard />', () => {
  it('Render Header', () => {
    const tree = renderer
      .create(
                    <FilterCard
                        filterAge={mockFunction}
                        filterCity={mockFunction}
                    />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
