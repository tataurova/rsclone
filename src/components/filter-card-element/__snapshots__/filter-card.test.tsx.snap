// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterCard /> Render Header 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <span
      className="MuiBox-root MuiBox-root-2"
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={true}
          htmlFor="outlined-number"
          id="outlined-number-label"
        >
          Age
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="outlined-number"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="30"
            required={false}
            type="number"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
            >
              <span>
                Age
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </span>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="outlined-search"
        id="outlined-search-label"
      >
        City
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          defaultValue=""
          disabled={false}
          id="outlined-search"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="search"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-5"
          >
            <span>
              City
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</form>
`;
